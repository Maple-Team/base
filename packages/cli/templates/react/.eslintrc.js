module.exports = {
  extends: '@liutsing/eslint-config',
  rules: {
    'multiline-ternary': 'off',
    indent: 'off',
    'operator-linebreak': 'off',
  },
  ignorePatterns: ['*.html', '*.json'],
}
