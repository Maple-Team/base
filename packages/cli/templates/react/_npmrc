auto-install-peers=true
strict-peer-dependencies=true
registry=https://registry.npmmirror.com