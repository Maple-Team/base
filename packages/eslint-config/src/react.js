/**
 * react 自定义规则配置
 * @type {import('eslint').Linter.RulesRecord}
 */
export const reactRules = {
  'jsx-quotes': ['error', 'prefer-double'],
  'react-hooks/rules-of-hooks': 'error',
  'react-hooks/exhaustive-deps': 'warn',
  'react/jsx-curly-newline': 'off',
  'react/jsx-indent': 'off',
  'react/jsx-handler-names': 'off',
  'react/prop-types': 'off',
  'react/display-name': 'off',
}
