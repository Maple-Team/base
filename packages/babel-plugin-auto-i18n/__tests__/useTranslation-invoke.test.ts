import { getTransformCode } from './test-help'
// pnpm run test .\__tests__\useTranslation-invoke.test.ts --coverage

describe('function declaration scenario', () => {
  describe('hooks scenarios', () => {
    it('case1', () => {
      const sourceCode = `
      import {useTranslation} from 'react-i18next'
      export function useComponent(){
        const var1 = "变量1"
        return [var1]
      }
      `
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('case2', () => {
      const sourceCode = `
      import {useTranslation} from 'react-i18next'
      export function useComponent(){
        const {i18n} = useTranslation()
        const var1 = "变量1"
        return [var1]
      }
      `
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('case3', () => {
      const sourceCode = `
      export function useComponent(){
        const {t} = useTranslation()
        const var1 = t("key1")
        return [var1]
      }
      `
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
  })
  describe('detect useTranslation invoke times', () => {
    it('case 1', () => {
      const sourceCode = `
      import {useTranslation} from 'react-i18next'
      export function useComponent(){
        return <div name="属性">站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useTranslation-invoke1.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('case 2', () => {
      const sourceCode = `
      import {useTranslation} from 'react-i18next'
      export function useComponent(){
        const {t, i18n} = useTranslation()
        return <div name="属性">站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useTranslation-invoke2.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('case 3', () => {
      const sourceCode = `
      import {useTranslation} from 'react-i18next'
      export function useComponent(){
        const {i18n} = useTranslation()
        return <div name="属性">站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useTranslation-invoke3.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('case 4', () => {
      const sourceCode = `
      import {useTranslation} from 'react-i18next'
      export function useComponent(){
        const {t} = useTranslation()
        return <div name="属性">{t('key')}</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useTranslation-invoke4.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('代码组件没有使用到中文则不调用useTranslation1', () => {
      const sourceCode = `
      import {useState} from 'react'
      export function useComponent(){
        const [v, setV] = useState(0)
        return {v,setV}
      }
      `
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('代码组件没有使用到中文则不调用useTranslation2', () => {
      const sourceCode = `
      import {useState} from 'react'
      export function Component(){
        const [v, setV] = useState(0)
        return <>{v}</>
      }
      `
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('代码组件没有使用到中文则不调用useTranslation3', () => {
      const sourceCode = `
      import {useState} from 'react'
      export function Component(){
        console.log('中文')
        const [v, setV] = useState(0)
        return <>{v}</>
      }
      `
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('代码组件没有使用到中文则不调用useTranslation4', () => {
      const sourceCode = `
      import {useState} from 'react'
      export function Component(){
        console.log('中文')
        const [v, setV] = useState(0)
        return <div>{v}</div>
      }
      `
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
  })
})

describe('arrow function scenario', () => {
  describe('hooks scenarios', () => {
    it('case1', () => {
      const sourceCode = `
      import {useTranslation} from 'react-i18next'
      export const useHooks1 = () => {
        const var1 = "变量1"
        return [var1]
      }
      `
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
  })
  // 确保useTranslation只调用一次
  describe('detect useTranslation invoke times', () => {
    it('case 1', () => {
      const sourceCode = `
      import {useTranslation} from 'react-i18next'
      export const Component = () => {
        return <div name="属性">站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useTranslation-invoke1.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('case 2', () => {
      const sourceCode = `
      import {useTranslation} from 'react-i18next'
      export const Component = () => {
        const {t, i18n} = useTranslation()
        return <div name="属性">站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useTranslation-invoke2.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('case 3', () => {
      const sourceCode = `
      import {useTranslation} from 'react-i18next'
      export const Component = () => {
        const {i18n} = useTranslation()
        return <div name="属性">站位文本</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useTranslation-invoke3.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('case 4', () => {
      const sourceCode = `
      import {useTranslation} from 'react-i18next'
      export const Component = () => {
        const {t} = useTranslation()
        return <div name="属性">{t('key')}</div>
      }
      `
      const result = getTransformCode(sourceCode, 'useTranslation-invoke4.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('代码组件没有使用到中文则不调用useTranslation1', () => {
      const sourceCode = `
      import {useState} from 'react'
      export const Component = () => {
        const [v, setV] = useState(0)
        return {v,setV}
      }
      `
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('代码组件没有使用到中文则不调用useTranslation2', () => {
      const sourceCode = `
      import {useState} from 'react'
      export const Component = () => {
        console.log('中文')
        const [v, setV] = useState(0)
        return {v,setV}
      }
      `
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('代码组件没有使用到中文则不调用useTranslation3', () => {
      const sourceCode = `
      import {useState} from 'react'
      export const Component = () => {
        console.log('中文')
        const [v, setV] = useState(0)
        return <>{v}</>
      }
      `
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('代码组件没有使用到中文则不调用useTranslation4', () => {
      const sourceCode = `
      import {useState} from 'react'
      export const Component = () => {
        console.log('中文')
        const [v, setV] = useState(0)
        return <div>{v}</div>
      }
      `
      const result = getTransformCode(sourceCode, 'filename.tsx')
      expect(result?.code).toMatchSnapshot()
    })
  })
})
