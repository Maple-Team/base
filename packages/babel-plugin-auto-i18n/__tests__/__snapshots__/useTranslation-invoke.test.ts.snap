// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrow function scenario detect useTranslation invoke times case 1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  return <div name={t(\\"key_属性\\")}>{t(\\"key_站位文本\\")}</div>;
};"
`;

exports[`arrow function scenario detect useTranslation invoke times case 2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  return <div name={t(\\"key_属性\\")}>{t(\\"key_站位文本\\")}</div>;
};"
`;

exports[`arrow function scenario detect useTranslation invoke times case 3 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    i18n = _useTranslation.i18n,
    t = _useTranslation.t;
  return <div name={t(\\"key_属性\\")}>{t(\\"key_站位文本\\")}</div>;
};"
`;

exports[`arrow function scenario detect useTranslation invoke times case 4 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  return <div name={t(\\"key_属性\\")}>{t('key')}</div>;
};"
`;

exports[`arrow function scenario detect useTranslation invoke times 代码组件没有使用到中文则不调用useTranslation1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var _react = require(\\"react\\");
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError(\\"Invalid attempt to destructure non-iterable instance.\\\\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.\\"); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === \\"string\\") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === \\"Object\\" && o.constructor) n = o.constructor.name; if (n === \\"Map\\" || n === \\"Set\\") return Array.from(o); if (n === \\"Arguments\\" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : \\"undefined\\" != typeof Symbol && r[Symbol.iterator] || r[\\"@@iterator\\"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t[\\"return\\"] && (u = t[\\"return\\"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
var Component = exports.Component = function Component() {
  var _useState = (0, _react.useState)(0),
    _useState2 = _slicedToArray(_useState, 2),
    v = _useState2[0],
    setV = _useState2[1];
  return {
    v: v,
    setV: setV
  };
};"
`;

exports[`arrow function scenario detect useTranslation invoke times 代码组件没有使用到中文则不调用useTranslation2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var _react = require(\\"react\\");
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError(\\"Invalid attempt to destructure non-iterable instance.\\\\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.\\"); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === \\"string\\") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === \\"Object\\" && o.constructor) n = o.constructor.name; if (n === \\"Map\\" || n === \\"Set\\") return Array.from(o); if (n === \\"Arguments\\" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : \\"undefined\\" != typeof Symbol && r[Symbol.iterator] || r[\\"@@iterator\\"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t[\\"return\\"] && (u = t[\\"return\\"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
var Component = exports.Component = function Component() {
  console.log('中文');
  var _useState = (0, _react.useState)(0),
    _useState2 = _slicedToArray(_useState, 2),
    v = _useState2[0],
    setV = _useState2[1];
  return {
    v: v,
    setV: setV
  };
};"
`;

exports[`arrow function scenario detect useTranslation invoke times 代码组件没有使用到中文则不调用useTranslation3 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var _react = require(\\"react\\");
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError(\\"Invalid attempt to destructure non-iterable instance.\\\\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.\\"); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === \\"string\\") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === \\"Object\\" && o.constructor) n = o.constructor.name; if (n === \\"Map\\" || n === \\"Set\\") return Array.from(o); if (n === \\"Arguments\\" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : \\"undefined\\" != typeof Symbol && r[Symbol.iterator] || r[\\"@@iterator\\"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t[\\"return\\"] && (u = t[\\"return\\"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
var Component = exports.Component = function Component() {
  console.log('中文');
  var _useState = (0, _react.useState)(0),
    _useState2 = _slicedToArray(_useState, 2),
    v = _useState2[0],
    setV = _useState2[1];
  return <>{v}</>;
};"
`;

exports[`arrow function scenario detect useTranslation invoke times 代码组件没有使用到中文则不调用useTranslation4 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var _react = require(\\"react\\");
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError(\\"Invalid attempt to destructure non-iterable instance.\\\\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.\\"); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === \\"string\\") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === \\"Object\\" && o.constructor) n = o.constructor.name; if (n === \\"Map\\" || n === \\"Set\\") return Array.from(o); if (n === \\"Arguments\\" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : \\"undefined\\" != typeof Symbol && r[Symbol.iterator] || r[\\"@@iterator\\"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t[\\"return\\"] && (u = t[\\"return\\"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
var Component = exports.Component = function Component() {
  console.log('中文');
  var _useState = (0, _react.useState)(0),
    _useState2 = _slicedToArray(_useState, 2),
    v = _useState2[0],
    setV = _useState2[1];
  return <div>{v}</div>;
};"
`;

exports[`arrow function scenario hooks scenarios case1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.useHooks1 = void 0;
var _reactI18next = require(\\"react-i18next\\");
var useHooks1 = exports.useHooks1 = function useHooks1() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  var var1 = t(\\"key_变量1\\");
  return [var1];
};"
`;

exports[`function declaration scenario detect useTranslation invoke times case 1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.useComponent = useComponent;
var _reactI18next = require(\\"react-i18next\\");
function useComponent() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  return <div name={t(\\"key_属性\\")}>{t(\\"key_站位文本\\")}</div>;
}"
`;

exports[`function declaration scenario detect useTranslation invoke times case 2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.useComponent = useComponent;
var _reactI18next = require(\\"react-i18next\\");
function useComponent() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  return <div name={t(\\"key_属性\\")}>{t(\\"key_站位文本\\")}</div>;
}"
`;

exports[`function declaration scenario detect useTranslation invoke times case 3 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.useComponent = useComponent;
var _reactI18next = require(\\"react-i18next\\");
function useComponent() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    i18n = _useTranslation.i18n,
    t = _useTranslation.t;
  return <div name={t(\\"key_属性\\")}>{t(\\"key_站位文本\\")}</div>;
}"
`;

exports[`function declaration scenario detect useTranslation invoke times case 4 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.useComponent = useComponent;
var _reactI18next = require(\\"react-i18next\\");
function useComponent() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  return <div name={t(\\"key_属性\\")}>{t('key')}</div>;
}"
`;

exports[`function declaration scenario detect useTranslation invoke times 代码组件没有使用到中文则不调用useTranslation1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.useComponent = useComponent;
var _reactI18next = require(\\"react-i18next\\");
var _react = require(\\"react\\");
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError(\\"Invalid attempt to destructure non-iterable instance.\\\\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.\\"); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === \\"string\\") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === \\"Object\\" && o.constructor) n = o.constructor.name; if (n === \\"Map\\" || n === \\"Set\\") return Array.from(o); if (n === \\"Arguments\\" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : \\"undefined\\" != typeof Symbol && r[Symbol.iterator] || r[\\"@@iterator\\"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t[\\"return\\"] && (u = t[\\"return\\"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function useComponent() {
  var _useState = (0, _react.useState)(0),
    _useState2 = _slicedToArray(_useState, 2),
    v = _useState2[0],
    setV = _useState2[1];
  return {
    v: v,
    setV: setV
  };
}"
`;

exports[`function declaration scenario detect useTranslation invoke times 代码组件没有使用到中文则不调用useTranslation2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = Component;
var _reactI18next = require(\\"react-i18next\\");
var _react = require(\\"react\\");
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError(\\"Invalid attempt to destructure non-iterable instance.\\\\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.\\"); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === \\"string\\") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === \\"Object\\" && o.constructor) n = o.constructor.name; if (n === \\"Map\\" || n === \\"Set\\") return Array.from(o); if (n === \\"Arguments\\" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : \\"undefined\\" != typeof Symbol && r[Symbol.iterator] || r[\\"@@iterator\\"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t[\\"return\\"] && (u = t[\\"return\\"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function Component() {
  var _useState = (0, _react.useState)(0),
    _useState2 = _slicedToArray(_useState, 2),
    v = _useState2[0],
    setV = _useState2[1];
  return <>{v}</>;
}"
`;

exports[`function declaration scenario detect useTranslation invoke times 代码组件没有使用到中文则不调用useTranslation3 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = Component;
var _reactI18next = require(\\"react-i18next\\");
var _react = require(\\"react\\");
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError(\\"Invalid attempt to destructure non-iterable instance.\\\\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.\\"); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === \\"string\\") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === \\"Object\\" && o.constructor) n = o.constructor.name; if (n === \\"Map\\" || n === \\"Set\\") return Array.from(o); if (n === \\"Arguments\\" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : \\"undefined\\" != typeof Symbol && r[Symbol.iterator] || r[\\"@@iterator\\"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t[\\"return\\"] && (u = t[\\"return\\"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function Component() {
  console.log('中文');
  var _useState = (0, _react.useState)(0),
    _useState2 = _slicedToArray(_useState, 2),
    v = _useState2[0],
    setV = _useState2[1];
  return <>{v}</>;
}"
`;

exports[`function declaration scenario detect useTranslation invoke times 代码组件没有使用到中文则不调用useTranslation4 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = Component;
var _reactI18next = require(\\"react-i18next\\");
var _react = require(\\"react\\");
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError(\\"Invalid attempt to destructure non-iterable instance.\\\\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.\\"); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === \\"string\\") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === \\"Object\\" && o.constructor) n = o.constructor.name; if (n === \\"Map\\" || n === \\"Set\\") return Array.from(o); if (n === \\"Arguments\\" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : \\"undefined\\" != typeof Symbol && r[Symbol.iterator] || r[\\"@@iterator\\"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t[\\"return\\"] && (u = t[\\"return\\"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function Component() {
  console.log('中文');
  var _useState = (0, _react.useState)(0),
    _useState2 = _slicedToArray(_useState, 2),
    v = _useState2[0],
    setV = _useState2[1];
  return <div>{v}</div>;
}"
`;

exports[`function declaration scenario hooks scenarios case1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.useComponent = useComponent;
var _reactI18next = require(\\"react-i18next\\");
function useComponent() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  var var1 = t(\\"key_变量1\\");
  return [var1];
}"
`;

exports[`function declaration scenario hooks scenarios case2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.useComponent = useComponent;
var _reactI18next = require(\\"react-i18next\\");
function useComponent() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    i18n = _useTranslation.i18n,
    t = _useTranslation.t;
  var var1 = t(\\"key_变量1\\");
  return [var1];
}"
`;

exports[`function declaration scenario hooks scenarios case3 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.useComponent = useComponent;
var _reactI18next = require(\\"react-i18next\\");
function useComponent() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t,
    i18n = _useTranslation.i18n;
  var var1 = t(\\"key1\\");
  return [var1];
}"
`;
